src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/cascaded_document_layout.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useMemo, useCallback, Fragment, useRef } from 'react';
+import { useEuiTheme } from '@elastic/eui';
+import { type AggregateQuery } from '@kbn/es-query';
+import { type Filter } from '@kbn/es-query';
+import {
+  DataCascade,
+  DataCascadeRow,
+  DataCascadeRowCell,
+  type DataCascadeRowCellProps,
+} from '@kbn/shared-ux-document-data-cascade';
+import type { UnifiedDataTableProps } from '@kbn/unified-data-table';
+import { getESQLStatsQueryMeta } from '@kbn/esql-utils';
+import { EsqlQuery } from '@kbn/esql-ast';
+import type { ESQLStatsQueryMeta } from '@kbn/esql-utils/src/utils/cascaded_documents_helpers';
+import { getStatsCommandToOperateOn } from '@kbn/esql-utils/src/utils/cascaded_documents_helpers';
+import { useDiscoverServices } from '../../../../../hooks/use_discover_services';
+import { useScopedServices } from '../../../../../components/scoped_services_provider/scoped_services_provider';
+import { useCurrentTabSelector, useAppStateSelector } from '../../../state_management/redux';
+import {
+  useEsqlDataCascadeRowHeaderComponents,
+  useEsqlDataCascadeHeaderComponent,
+  ESQLDataCascadeLeafCell,
+  type ESQLDataGroupNode,
+  type DataTableRecord,
+} from './blocks';
+import { cascadedDocumentsStyles } from './cascaded_documents.styles';
+import { type CascadedDocumentsRestorableState } from './cascaded_documents_restorable_state';
+import { useEsqlDataCascadeRowActionHelpers } from './blocks/use_row_header_components';
+import {
+  useDataCascadeRowExpansionHandlers,
+  useGroupedCascadeData,
+  useScopedESQLQueryFetchClient,
+} from './hooks';
+import type { DiscoverStateContainer } from '../../../state_management/discover_state';
+
+export { getESQLStatsQueryMeta };
+export { useGetGroupBySelectorRenderer as useGroupBySelectorRenderer } from './blocks/use_table_header_components';
+
+export interface ESQLDataCascadeProps extends Omit<UnifiedDataTableProps, 'ref'> {
+  defaultFilters?: Filter[];
+  viewModeToggle: React.ReactElement | undefined;
+  cascadeGroupingChangeHandler: (cascadeGrouping: string[]) => void;
+  cascadeConfig: CascadedDocumentsRestorableState;
+  togglePopover: ReturnType<typeof useEsqlDataCascadeRowActionHelpers>['togglePopover'];
+  queryMeta: ESQLStatsQueryMeta;
+}
+
+const ESQLDataCascade = React.memo(
+  ({
+    rows: initialData,
```

Comment:
```suggestion
    rows,
```
Looks like we only pass it as `rows` below too, so can probably skip the rename.